tap_slack/client.py

```diff
@@ -1,18 +1,40 @@
 """REST client handling, including SlackStream base class."""
 
+from __future__ import annotations
+
 import time
-from typing import Any, Dict, Text, Optional, List
+from typing import Any, Dict, List, Optional, Text
 
-from singer_sdk.streams import RESTStream
+from requests import Response
 from singer_sdk.authenticators import BearerTokenAuthenticator
+from singer_sdk.pagination import JSONPathPaginator
+from singer_sdk.streams import RESTStream
+
+
+class ThrottledJSONPathPaginator(JSONPathPaginator):
```

Comment:
@kgpayne I'm assuming this is to keep status quo based on new pagination logic in the SDK, is that correct?

Reply:
@pnadolny13 correct 🙂 IIRC I had an issue with the existing paginator that was resolved by reimplementing the throttling with a new-style paginator. However the logic didn’t change 👍

Reply:
@kgpayne awesome, thanks for clarifying! I also tested in Squared and it works as expected.